File: Python/PowerSupply.py
# power_supply.py
from Komponen import Komponen

class PowerSupply(Komponen):
    """Power Supply class"""
    
    def __init__(self, wattage=0, certification="", merk="", nama="", harga=0, tipe_komponen="Power Supply"):
        super().__init__(merk, nama, harga, tipe_komponen)
        self.wattage = wattage
        self.certification = certification
    
    def get_wattage(self):
        return self.wattage
    
    def set_wattage(self, wattage):
        self.wattage = wattage
    
    def get_certification(self):
        return self.certification
    
    def set_certification(self, certification):
        self.certification = certification
    
    def tampilkan_info(self):
        print(f"Power Supply: {self.merk} {self.nama} ({self.wattage}W, {self.certification})")
        print(f"    Harga: Rp{self.harga}")

File: Java/PowerSupply.java
public class PowerSupply extends Komponen {
    private int wattage;
    private String certification;
    
    public PowerSupply() {
        super();
        this.wattage = 0;
        this.certification = "";
    }
    
    public PowerSupply(int wattage, String certification, String merk, String nama, double harga) {
        super(merk, nama, harga, "Power Supply");
        this.wattage = wattage;
        this.certification = certification;
    }
    
    public int getWattage() {
        return wattage;
    }
    
    public void setWattage(int wattage) {
        this.wattage = wattage;
    }
    
    public String getCertification() {
        return certification;
    }
    
    public void setCertification(String certification) {
        this.certification = certification;
    }
    
    @Override
    public void tampilkanInfo() {
        System.out.println("Power Supply: " + merk + " " + nama + 
                           " (" + wattage + "W, " + certification + ")");
        System.out.println("    Harga: Rp" + harga);
    }
}
